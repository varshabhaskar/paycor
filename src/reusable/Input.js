import React from "react";
import PropTypes from "prop-types";
import { errorColor } from "../styles";

const errorStyle = {
  color: errorColor,
  fontWeight: "bold"
};
const inputErrorStyle = {
  border: "solid 1px " + errorColor
};
function Input(props) {
  return (
    <div>
      <label htmlFor={props.id}>{props.label}</label>
      <br />
      <input
        id={props.id}
        type={props.type}
        name={props.name} // this is the property we wanna set onChange. can name it anything
        value={props.value}
        onChange={props.onChange}
        style={props.error && inputErrorStyle}
      ></input>
      {props.error && <p style={errorStyle}>{props.error}</p>}
    </div>
  );
}

Input.propTypes = {
  id: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  type: PropTypes.oneOf(["text", "email", "phone", "number"]),
  name: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.string.isRequired
};

export default Input;
