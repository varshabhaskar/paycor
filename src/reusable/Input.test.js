import Input from "./Input";
import React from "react";
import { render } from "@testing-library/react";
import renderer from "react-test-renderer";
it("should display when passed using getByText", () => {
  //arrange
  const { getByText } = render(
    <Input
      id="test"
      name="test"
      label="test"
      onChange={() => {}}
      value="test"
      error="exampleerror"
    />
  );
  //act
  //assert
  getByText("exampleerror");
});

it("should display when passed using getByRole", () => {
  //arrange
  const { getByRole } = render(
    <Input
      id="test"
      name="test"
      label="test"
      onChange={() => {}}
      value="test"
      error="example error"
    />
  );
  //act
  //assert
  const error = getByRole("alert");
  expect(error.innerHTML).toBe("example error");
});

it("should not render the error para when error is not provided", () => {
  //arrange
  //act
  const tree = renderer
    .create(
      <Input
        id="test"
        name="test"
        label="test"
        onChange={() => {}}
        value="test"
      />
    )
    .toJSON();
  //assert
  expect(tree).toMatchSnapshot();
});
