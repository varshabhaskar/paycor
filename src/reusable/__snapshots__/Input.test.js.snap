// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render the error para when error is not provided 1`] = `
<div>
  <label
    htmlFor="test"
  >
    test
  </label>
  <br />
  <input
    id="test"
    name="test"
    onChange={[Function]}
    value="test"
  />
</div>
`;
